### PR TITLE
Revert HTML CW changes

### PR DESCRIPTION
**Pull request by beatrix-bitrot**
set spoiler_text back to just be text, the way it was before. clients will have to link hashtags themselves if they want!

partly reverts changes from #2970 
completely reverts changes from #2683

### CODE REVIEW

File: app/javascript/mastodon/components/status_content.js [hunk 1]
```diff
@@ -68,11 +70,11 @@ class StatusContent extends React.PureComponent {
     const [ startX, startY ] = this.startXY;
     const [ deltaX, deltaY ] = [Math.abs(e.clientX - startX), Math.abs(e.clientY - startY)];
 
-    if (e.target.localName === 'button' || e.target.localName === 'a' || (e.target.parentNode && e.target.parentNode.localName === 'a')) {
+    if (e.target.localName === 'a' || (e.target.parentNode && e.target.parentNode.localName === 'a')) {
```
Comment (Gargron) [inline app/javascript/mastodon/components/status_content.js]: We wanna keep this and 77

File: app/javascript/mastodon/components/status_content.js [hunk 2]
```diff
@@ -36,6 +36,8 @@ class StatusContent extends React.PureComponent {
         link.setAttribute('title', mention.get('acct'));
       } else if (link.textContent[0] === '#' || (link.previousSibling && link.previousSibling.textContent && link.previousSibling.textContent[link.previousSibling.textContent.length - 1] === '#')) {
         link.addEventListener('click', this.onHashtagClick.bind(this, link.text), false);
+      } else if (media) {
```
Comment (Gargron) [inline app/javascript/mastodon/components/status_content.js]: This should stay gone

File: app/javascript/mastodon/components/status_content.js [hunk 3]
```diff
@@ -116,19 +118,14 @@ class StatusContent extends React.PureComponent {
       }
 
       return (
-        <div
-          ref={this.setRef}
-          className='status__content'
-          onMouseDown={this.handleMouseDown}
-          onMouseUp={this.handleMouseUp}
-        >
+        <div className='status__content' onMouseDown={this.handleMouseDown} onMouseUp={this.handleMouseUp}>
```
Comment (Gargron) [inline app/javascript/mastodon/components/status_content.js]: All of below can stay as they were

File: app/lib/atom_serializer.rb [hunk 4]
```diff
@@ -336,8 +336,8 @@ def add_namespaces(parent)
   end
 
   def serialize_status_attributes(entry, status)
-    append_element(entry, 'summary', Formatter.instance.format(status.proper, :spoiler_text, false).to_str, 'xml:lang': status.language, type: 'html') if status.spoiler_text?
     append_element(entry, 'content', Formatter.instance.format(status).to_str, type: 'html', 'xml:lang': status.language)
+    append_element(entry, 'summary', status.spoiler_text, 'xml:lang': status.language) if status.spoiler_text?
```
Comment (Gargron) [inline app/lib/atom_serializer.rb]: Doesn't really matter but I liked summary be before content. Otherwise correct

**Review by Gargron — CHANGES_REQUESTED** [review]

File: app/javascript/mastodon/components/status_content.js (hunk 1 above)
Comment (beatrix-bitrot) [inline app/javascript/mastodon/components/status_content.js]: are you sure? cause we introduced a regression where clicking the CW button would also open the detailed status and after this changeset it is fixed, which I suspect is related to this  particular change though i didn't test it in isolation

Comment (Gargron) [inline app/javascript/mastodon/components/status_content.js]: I couldn't reproduce that behaviour, it shouldn't depend on button vs a anyway, but button is more a11y-friendly

Comment (beatrix-bitrot) [inline app/javascript/mastodon/components/status_content.js]: agreed ok, will do so

**Review by Gargron — APPROVED** [review]

**Pull request merged**